JS/Lesson19/saffet/exercises.js

```diff
@@ -1,8 +1,201 @@
+class Person {
+    constructor(name, age, gender){
+        this.name = name;
+        this.age = age;
+        this.gender = gender;
+    }
 
-// 3. Create a Vehicle class with make, model, and year properties, and a start() method that logs a message to the console indicating that the vehicle has started. Then create a Car class that extends the Vehicle class and adds a numDoors property. Override the start() method in the Car class to log a different message to the console indicating that the car has started.
+    greet(){
+        console.log(`Greetings from ${this.name}`);
+    }
+}
 
+class Student extends Person{
+    constructor(name, age, gender, major){
+        super(name, age, gender);
+        this.major = major;
+    }
 
-// 4. Create a BankAccount class with balance and interestRate properties, and deposit() and withdraw() methods. Add a transactionHistory property that is an array of objects representing each transaction made on the account, with properties for type (either "deposit" or "withdrawal"), amount, and date ({type: "withdrawal", amount: amount, date: new Date().toLocaleDateString()}). Implement deposit() and withdraw() methods that add new transactions to the transactionHistory array. Implement a get transactionHistory() getter method that returns a copy of the transaction history array, and a get currentBalance() getter method that calculates and returns the current balance of the account based on the transaction history.
+    greet(){
+        console.log(`Greetings from ${this.name}, a student who is studying ${this.major}`);
+    }
+}
+
+const person = new Person("Saffet", 26, "Male");
+const student = new Student("Ramazan", 30, "Male", "Electrical&Electronics Engineering");
+
+console.log(person);
+console.log(student);
+
+console.log("==================================");
+// 2. Create a Shape class with name and color properties, and a describe() method that logs a description of the shape to the 
+// console (e.g. "This is a blue square"). Then create a Square class that extends the Shape class and adds sideLength and area 
+// properties. Override the describe() method in the Square class to log a description of the square to the console 
+// (e.g. "This is a blue square with a side length of 5 and an area of 25").
+console.log("Example 2; ");
+
+class Shape{
+    constructor(name, color){
+        this.name = name;
+        this.color = color;
+    }
+
+    describe(){
+        console.log(`This is a ${this.color} ${this.name}`);
+    }
+}
+
+class Square extends Shape{
+    constructor(name = "Square", color, sideLenght){
+        super(color);
```

Comment:
You need to pass both `name` and `color` to the `super` method, without it you get "undefined" instead of the color in the message "This is a undefined Square with a side lenght of 4 and area of 16".